Python/pywarpx/_libwarpx.py

```diff
@@ -18,10 +18,15 @@
 
 try:
     # --- If mpi4py is going to be used, this needs to be imported
-    # --- before libwarpx is loaded (though don't know why)
+    # --- before libwarpx is loaded, because mpi4py calls MPI_Init
     from mpi4py import MPI
+    if MPI._sizeof(MPI.Comm) == ctypes.sizeof(ctypes.c_int):
+        _MPI_Comm_type = ctypes.c_int
+    else:
+        _MPI_Comm_type = ctypes.c_void_p
```

Comment:
Is it ok if we add an inline code comment here?
If I see the code correctly, you are switching essentially between OpenMPI and MPICH flavors based on the type of the communicator? Is that correct? Let's add a few lines here to conserve that knowledge for future readers? :)